pkg/mounter/helper.go

```diff
@@ -86,3 +92,49 @@ func ValidateAnnotationsSize(annotations map[string]string) error {
 	}
 	return nil
 }
+
+// GetClusterId get clusterId from env or profile for RRSA
+func GetClusterId() (clusterId string, err error) {
+	clusterId = os.Getenv("CLUSTER_ID")
+	if clusterId != "" {
+		return
+	}
+	cfg, err := clientcmd.BuildConfigFromFlags(options.MasterURL, options.Kubeconfig)
+	if err != nil {
+		return
+	}
+	clientset, err := kubernetes.NewForConfig(cfg)
+	if err != nil {
+		return
+	}
+	profile, err := clientset.CoreV1().ConfigMaps("kube-system").Get(context.Background(), "ack-cluster-profile", metav1.GetOptions{})
+	if err != nil {
+		return
+	}
+	clusterId = profile.Data["clusterid"]
+	return
+}
+
+// GetOIDCProvider get OIDC provider from env or ACK clusterId for RRSA
+func GetOIDCProvider() (provider string, err error) {
+	provider = os.Getenv("OIDC_PROVIDER")
```

Comment:
I still would like this one to override the full ARN, and can be called "ALIBABA_CLOUD_OIDC_PROVIDER_ARN" for consistency. Can't a user potentially use a provider from others account?

And should this be a parameter in storage class? Maybe we can support multiple providers for different PV, or even different CSI drivers. Or CSI itself may gain OIDC capability in the future. Put it here can be ambiguous.

Reply:
If role_arn or provider_arn does not conform to this fixed format, it means that the user is trying to access the OSS from other cloud vendor through RRSA. But ossfs itself does not have the ability to parse the returns of STS services from other cloud vendors. CSI can accept a full arn but it will still not work in ossfs.  

And for multiple providers, I think supporting multiple roles is enough. @mowangdk PTAL

Reply:
I mean the user may also want to override the account ID part of ARN. Consider that user A may want to access an OSS bucket from account B. User B then adds the ACK cluster in account A to his OIDC provider, then user A will need to specify the provider from account B.

Reply:
sts will return error like 
![image](https://github.com/kubernetes-sigs/alibaba-cloud-csi-driver/assets/83764662/f21ab83f-833b-4768-8518-a5812c02de20)

Reply:
I've tested that I can register ACK cluster in account A to OIDC provider in account B. Then everything else should only relate to account B. Which step (which OpenAPI) would fail?

Reply:
could you please give a detailed test case and the AssumeRoleWithOIDC related logs of ossfs?

Reply:
I don't know much about ossfs. But I've tried my proposed flow, it works and I got token from AssumeRoleWithOIDC.
1. Login account A, create an ACK cluster, enable RRSA, copy the provider URL
2. Create a Pod in above ACK cluster, mount this volume, and copy the token in it.
   ```yaml
   - name: rrsa-oidc-token
     projected:
       defaultMode: 0600
       sources:
       - serviceAccountToken:
           audience: sts.aliyuncs.com
           expirationSeconds: 3600
           path: token
   ```
3. Login account B, go to https://ram.console.aliyun.com/providers/create/OIDC and register an OIDC provider with URL from step 1, copy the provider ARN
4. create a RAM role for OIDC, copy the role ARN
5. go to https://next.api.aliyun.com/api/Sts/2015-04-01/AssumeRoleWithOIDC, fill in token from step 2, and ARNs from step 3 and 4, I get RAM sts token.

These steps should naturally apply to ossfs.